pkg/controller/plan/kubevirt.go

```diff
@@ -1272,6 +1272,10 @@ func (r *KubeVirt) vmPreference(vm *plan.VMStatus) (virtualMachine *cnv.VirtualM
 	if err != nil {
 		return
 	}
+	if preferenceName == "" {
+		err = liberr.New("couldn't find the right OS.", "vm", vm)
```

Comment:
please rephrase that message, it should probably say something like "couldn't find a corresponding preference"

Reply:
that semi-true. we saw it when the incoming VM had the OS field missing, probably template will fail as well.
since it might be on different cases as well (os that is not in the config map), it does make sense to switch it into preference case message :)